source/speech/speech.py

```diff
@@ -774,7 +774,8 @@ def _objectSpeech_calculateAllowedProps(reason, shouldReportTextContent):
 	}
 	if reason in (OutputReason.FOCUSENTERED, OutputReason.MOUSE):
 		allowProperties["value"] = False
-		allowProperties["keyboardShortcut"] = False
+		if not objRole == controlTypes.Role.LIST:

```

Comment:
> I think we should speak shortcuts for any object that focus enters. I cannot remember specifically why we never originally did. If you can make an argument for lists, I think you can for any other container. Most likely this was simply that early on it was only done for focus and not focusEntered, and then when later abstracted, it was deliberately removed from focusEntered to match existing behaviour.

@michaelDCurran, this has just been discussed with @leonardder in https://github.com/nvaccess/nvda/pull/15826#issuecomment-1826054845, https://github.com/nvaccess/nvda/pull/15826#issuecomment-1826121542 and https://github.com/nvaccess/nvda/pull/15826#issuecomment-1826355984.

The group container can have a shortcut key defined but actually not working. To avoid to get other such not working case, it seems safer to restrict this PR to the only known, frequent and useful case of lists. If in the future we can find other containers for which the shortcut key should be announced, we may include their role too.

Or should I instead exclude explicitly the case of grouping which is not working?

A third possibility can be to remove this filtering and consider that putting a shortcut key in grouping where it does not work should not be done in first place by the developer of the application.

@michaelDCurran, let me know which solution you would prefer.

Reply:
Whether authors place a shortcut on groupings, or whether these shortcuts actually work on groupings in applications really is separate to this pr I feel. However, your current solution solves your immediate problem, and does not introduce other known problems. Therefore, could you please at least add a comment above this line stating why we only do it for list at this stage.

Reply:
OK. Done in 43c500e628ef2ef9d79be0abd65081b8a6fcd0cb.